source/client/Controller.py

```diff
@@ -83,33 +90,38 @@ def draw(self):
         #Transition phase immediately to avoid double draw
         self._state.turn_phase = Turn_Phases[3]
     
-    def pickUpPile(self):
+    def pickUpPile(self, note):
         """Attempt to pick up the pile"""
         if self._state.turn_phase != Turn_Phases[1]:
-            self.note = "You can only pick up the pile at the start of your turn"
+            self.note = note
             return
         try:
             self._state.pickupPileRuleCheck(self.prepared_cards)
         except Exception as err:
             self.note = "{0}".format(err)
         else:
-            self._state.turn_phase = Turn_Phases[2] #Set turn phase to reflect forced action
-            self.note = "Waiting for new cards to make required play"
+            if self._state.rules.play_pick_up:
+                self._state.turn_phase = Turn_Phases[2] #Set turn phase to reflect forced action
+                self.note = "Waiting for new cards to make required play"
+            else:
+                self._state.turn_phase = Turn_Phases[3] # No action required if rules.play_pick_up = False
             connection.Send({"action": "pickUpPile"})
 
     def makeForcedPlay(self, top_card):
-        """Complete the required play for picking up the pile"""
+        """Complete the required play for picking up the pile, (used in HandAndFoot but not Liverpool)"""
         self.note = "Performing the play required to pick up the pile"
         #Get key for top_card (we know it can be auto-keyed), and then prepare it
         key = self._state.getValidKeys(top_card)[0]
-        self.prepared_cards.setdefault(key, []).append(top_card) #Can't just call prepared card b/c of turn phase checking
+        self.prepared_cards.setdefault(key, []).append(top_card)
+        #Can't just call prepared card b/c of turn phase checking
```

Comment:
This comment goes with the line above it. If you are going to take comments out of being at the end of the codeline they should go above the line of code they are about not below.